src/metabase/driver/sql/query_processor/util.clj

```diff
@@ -0,0 +1,21 @@
+(ns metabase.driver.sql.query-processor.util
+  (:require [metabase.util.honeysql-extensions :as hx]))
+
+(defn nfc-field->parent-identifier
+  "Take a nested field column field corresponding to something like an inner key within a JSON column,
+  and then get the parent column's identifier from its own identifier and the nfc path stored in the field.
+
+  Suppose you have the child with corresponding identifier
+
+  (metabase.util.honeysql-extensions/identifier :field \"blah -> boop\")
+
+  Ultimately, this is just a way to get the parent identifier
+
+  (metabase.util.honeysql-extensions/identifier :field \"blah\")"
+  [field-identifier field]
+  (let [nfc-path          (:nfc_path field)
+        parent-components (-> (:components field-identifier)
+                              (vec)
+                              (pop)
+                              (conj (first nfc-path)))]
+    (apply hx/identifier (cons :field parent-components))))
```

Comment:
This was in `metabase.models.field`, but it's only used to implement a `:sql` QP JSON support method for the `:postgres` and `:mysql` drivers. Since the QP/drivers are still using Honey SQL 1 this was making it hard for me to enforce a no-HoneySQL-1 rule in Kondo outside of QP/drivers code. So I moved it into a `:sql` driver util namespace